drivers/net/w5500.c

```diff
@@ -0,0 +1,2326 @@
+
+  w5500_write8(self,
+               W5500_BSB_SOCKET_REGS(0),
+               W5500_SN_TXBUF_SIZE,
+               SN_TXBUF_16KB);
+
+  /* ... and none to sockets 1 to 7. */
+
+  for (i = 1; i < 8; i++)
+    {
+      w5500_write8(self,
+                   W5500_BSB_SOCKET_REGS(i),
+                   W5500_SN_RXBUF_SIZE,
+                   SN_RXBUF_0KB);
+
+      w5500_write8(self,
+                   W5500_BSB_SOCKET_REGS(i),
+                   W5500_SN_TXBUF_SIZE,
+                   SN_TXBUF_0KB);
+    }
+
+  /* Enable RECV interrupts on socket 0 (SEND_OK interrupts will only be
+   * enabled as long as a transmission is in progress).
+   */
+
+  w5500_write8(self,
+               W5500_BSB_SOCKET_REGS(0),
+               W5500_SN_IMR,
+               SN_INT_RECV);
+
+  /* Enable interrupts on socket 0 */
+
+  w5500_write8(self,
+               W5500_BSB_COMMON_REGS,
+               W5500_SIMR, /* Socket Interrupt Mask Register */
+               SIMR(0));
+
+  /* Open socket 0 */
+
+  w5500_write8(self,
+               W5500_BSB_SOCKET_REGS(0),
+               W5500_SN_CR, /* Control Register */
+               SN_CR_OPEN);
+
+  /* Check wheter socket 0 is open in MACRAW mode. */
+
+  value = w5500_read8(self,
+                      W5500_BSB_SOCKET_REGS(0),
+                      W5500_SN_SR);
+
+  if (value != SN_SR_SOCK_MACRAW)
+    {
+      nerr("Unexpected status: %02" PRIx8 "\n", value);
+      goto error;
+    }
+
+  /* Reset Tx buffer management state. */
+
+  w5500_txbuf_reset(self);
+
+  /* Wait up to 10 seconds for link-up */
+
+  value = w5500_read8(self,
+                      W5500_BSB_COMMON_REGS,
+                      W5500_PHYCFGR);
+
+  for (i = 0; (i < 100) && !(value & PHYCFGR_LNK); i++)
+    {
+      value = w5500_read8(self,
+                          W5500_BSB_COMMON_REGS,
+                          W5500_PHYCFGR);
+
+      nxsig_usleep(100000); /* 100 ms x 100 = 10 sec */
+    }
+
+  if (value & PHYCFGR_LNK)
+    {
+      ninfo("Link up (%d Mbps / %s duplex)\n",
+            (value & PHYCFGR_SPD) ? 100 : 10,
+            (value & PHYCFGR_DPX) ? "full" : "half");
+    }
+  else
+    {
+      nwarn("Link still down.  Cable plugged?\n");
+      goto error;
+    }
+
+  return OK;
+
+error:
+  w5500_fence(self);
+  return -EIO;
+}
+
+/****************************************************************************
+ * Name: w5500_transmit
+ *
+ * Description:
+ *   Start hardware transmission.  Called either from the txdone interrupt
+ *   handling or from watchdog based polling.
+ *
+ * Input Parameters:
+ *   priv - Reference to the driver state structure
+ *
+ * Returned Value:
+ *   OK on success; a negated errno on failure
+ *
+ * Assumptions:
+ *   The network is locked.
+ *
+ ****************************************************************************/
+
+static void w5500_transmit(FAR struct w5500_driver_s *self)
+{
+  uint16_t offset;
+
+  /* Verify that the hardware is ready to send another packet.  If we get
```

Comment:
I am not so sure about these.  Isn't it normal in written english to put two spaces after a period?  That is for example the case in the ASF license text and the drivers/net/skeleton.c template also does it on all comments.  Or a I missing something?

Reply:
Good question, I suggested it because I saw other developers suggesting this modification in other PRs, but it is just a suggestion. Feel free to keep it.